compiler/qsc_qasm3/src/semantic/tests/statements/continue_stmt.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::semantic::tests::check_stmt_kinds;
+use expect_test::expect;
+
+#[test]
+fn for_loop() {
+    check_stmt_kinds(
+        "for int i in {1, 2, 3} continue;",
+        &expect![[r#"
+            ForStmt [0-32]:
+                loop_variable: 8
+                iterable: DiscreteSet [13-22]:
+                    values:
+                        Expr [14-15]:
+                            ty: Int(None, true)
+                            kind: Lit: Int(1)
+                        Expr [17-18]:
+                            ty: Int(None, true)
+                            kind: Lit: Int(2)
+                        Expr [20-21]:
+                            ty: Int(None, true)
+                            kind: Lit: Int(3)
+                body: Stmt [23-32]:
+                    annotations: <empty>
+                    kind: ContinueStmt [23-32]:
+        "#]],
+    );
+}
+
+#[test]
+fn while_loop() {
+    check_stmt_kinds(
+        "while (true) continue;",
+        &expect![[r#"
+            WhileLoop [0-22]:
+                condition: Expr [7-11]:
+                    ty: Bool(true)
+                    kind: Lit: Bool(true)
+                body: Stmt [13-22]:
+                    annotations: <empty>
+                    kind: ContinueStmt [13-22]:
+        "#]],
+    );
+}
+
+#[test]
+fn nested_scopes() {
+    check_stmt_kinds(
+        "while (true) { { continue; } }",
+        &expect![[r#"
+            WhileLoop [0-30]:
+                condition: Expr [7-11]:
+                    ty: Bool(true)
+                    kind: Lit: Bool(true)
+                body: Stmt [13-30]:
+                    annotations: <empty>
+                    kind: Block [13-30]:
+                        Stmt [15-28]:
+                            annotations: <empty>
+                            kind: Block [15-28]:
+                                Stmt [17-26]:
+                                    annotations: <empty>
+                                    kind: ContinueStmt [17-26]:
+        "#]],
+    );
+}
+
+#[test]
+fn continue_in_non_loop_scope_fails() {
+    check_stmt_kinds(
+        "continue;",
+        &expect![[r#"
+            Program:
+                version: <none>
+                statements:
+                    Stmt [0-9]:
+                        annotations: <empty>
+                        kind: Err
+
+            [Qsc.Qasm3.Compile.InvalidScope
+
+              x continue can only appear in loop scopes.
```

Comment:
this message is onconsistent with braek error of the same time. Do they have different legal locations?